src/uct/ib/base/ib_device.c

```diff
@@ -549,12 +552,135 @@ ucs_status_t uct_ib_device_query(uct_ib_device_t *dev,
     return UCS_OK;
 }
 
+#if defined(__x86_64__)
+static int uct_ib_device_cleanup_proc(void* arg) {
+    uct_ib_device_nb_close_ctx *ctx = arg;
+    static const char *process_name = "ucx_cleanup";
+    struct rlimit nofile;
+    int ret = -EINTR;
+    int fd;
+    char c;
+
+    prctl(PR_SET_NAME, (uintptr_t)&process_name, 0, 0, 0);
+
+    getrlimit(RLIMIT_NOFILE, &nofile);
+    for (fd = 0; fd < nofile.rlim_cur && !ctx->quit; fd++) {
+        if ((fd != ctx->cmd_fd) && (fd != ctx->pipefds[0])) {
+            ucs_x86_syscall(SYS_close, fd);
+        }
+    }
+
+    /* loop until pipe closed - either parent terminated or closing device */
+    while (!ctx->quit && ret == -EINTR) {
+        ret = ucs_x86_syscall(SYS_read, ctx->pipefds[0], &c, 1);
+    }
+
+    return 0;
+}
+#endif
+
+static ucs_status_t uct_ib_device_init_nb_close_ctx(uct_ib_device_t *dev)
+{
+#if defined(__x86_64__)
+    const int stack_size = 32 * 1024;
+    const int tls_size = 1024;
```

Comment:
1. align on =
2. stack_size = 4 * UCS_KBYTE
3. tls_size - can we take from some glibc func?
4. do we really need to define these constatnts? already have vars in ctx->